### PR TITLE
attempt at setting some more test arguments correctly

### PR DESCRIPTION
**Pull request by alpmestan**
This patch adds 2 more packages (both required for at least some tests) and tries to set some more test options correctly, using the information we have from the `Flavour`. Feedback welcome! I'm not sure how reliable that is.

### CODE REVIEW

File: src/GHC.hs [hunk 1]
```diff
@@ -107,6 +107,8 @@ stage2Packages = return [haddock]
 testsuitePackages :: Action [Package]
 testsuitePackages = return [ checkApiAnnotations
                            , checkPpr
+                           , ghcPkg
```
Comment (chitrak7) [inline src/GHC.hs]: We have ghcPkg in stage1. Do we need it here also?

Comment (alpmestan) [reply src/GHC.hs]: Well, it was not getting built here with `build.sh [...] test`. After adding it there, it was.

File: src/Rules/Test.hs [hunk 2]
```diff
@@ -71,7 +71,12 @@ needTestsuiteBuilders = do
     need targets
   where
     needfile :: Stage -> Package -> Action FilePath
-    needfile stage pkg = programPath =<< programContext stage pkg
+    needfile stage pkg
+      -- TODO (Alp): we might sometimes need more than vanilla!
+      -- This should therefore depend on what test ways
+      -- we are going to use, I suppose?
```
Comment (chitrak7) [inline src/Rules/Test.hs]: I am currently working on this. Think a PR will be ready in shortly.

Comment (alpmestan) [reply src/Rules/Test.hs]: Good. Thanks!

File: src/GHC.hs (hunk 1 above)
Comment (chitrak7) [reply src/GHC.hs]: In this case, maybe there are other stage1 packages that are not built properly. Should we change the need rule implemented there to need all stage1 packages?

**Comment by alpmestan** [comment]
This is what I get with this patch:

```
SUMMARY for test run started at Fri Jun 15 11:50:07 2018 CEST
 0:15:43 spent to go through
    6416 total tests, which gave rise to
   23725 test cases, of which
   16974 were skipped

      40 had missing libraries
    5714 expected passes
      92 expected failures

      67 caused framework failures
       0 caused framework warnings
       1 unexpected passes
     896 unexpected failures
       1 unexpected stat failures
```

Back when I tried #621, the results that I got were:

```
SUMMARY for test run started at Thu Jun 14 13:24:13 2018 CEST
 0:12:38 spent to go through
    6416 total tests, which gave rise to
   15870 test cases, of which
    9186 were skipped

      36 had missing libraries
    5627 expected passes
     114 expected failures

      63 caused framework failures
       0 caused framework warnings
      54 unexpected passes
```

So we're skipping **a lot less** tests now, and I could have increased that number I suppose if we were building `dyn` etc variants of things by default here. And shouldn't we be? The default flavour (`perf`) surely builds a whole bunch of things.

File: src/Rules/Test.hs [hunk 3]
```diff
@@ -71,7 +71,12 @@ needTestsuiteBuilders = do
...
+      | isLibrary pkg = pkgConfFile (Context stage pkg (read "v"))
```
Comment (snowleopard) [inline src/Rules/Test.hs]: Shall we change `Context stage pkg (read "v")` to `vanillaContext stage pkg`?

File: src/GHC.hs (hunk 1 above)
Comment (snowleopard) [reply src/GHC.hs]: I think `testsuitePackages` should list all packages that are requried for the testsuite. It doesn't matter what we build in Stage0 or Stage1.

**Comment by snowleopard** [comment]
@alpmestan Thanks! Looks good, so happy to merge. Just a minor comment above.

File: src/Rules/Test.hs (hunk 3 above)
Comment (alpmestan) [reply src/Rules/Test.hs]: Oh, yes, right.

File: src/GHC.hs (hunk 1 above)
Comment (alpmestan) [reply src/GHC.hs]: Yeah I think it's fine to put those in `testsuitePackages`, as at least for the two ones that I added, they actually are (required and) used by some tests.

**Comment by alpmestan** [comment]
>  Just a minor comment above

Addressed :-)

**Comment by snowleopard** [comment]
Thanks, merged!

**Pull request merged**